src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -4305,9 +4305,10 @@ public static VarHandle byteArrayViewVarHandle(Class<?> viewArrayClass,
      * If access is aligned then following access modes are supported and are
      * guaranteed to support atomic access:
      * <ul>
-     * <li>read write access modes for all {@code T}, with the exception of
-     *     access modes {@code get} and {@code set} for {@code long} and
-     *     {@code double} on 32-bit platforms.
+     * <li>read write access modes for all {@code T}. On 32-bit platforms,
+     *     access modes {@code get} and {@code set} for {@code long}, {@code
+     *     double} are supported but might lead to word tearing, as described in
+     *     Section {@jls 17.7} of <cite>The Java Language Specification</cite>.
```

Comment:
...except that JLS 17.7 is "Non-Atomic Treatment of double and long". Word tearing (JLS 17.6) should still be very much forbidden. This all means that `[MemoryLayout.java](https://github.com/openjdk/jdk/blob/ee0d309bbd33302d8c6f35155e975db77aaea785/src/java.base/share/classes/java/lang/foreign/MemoryLayout.java#L279-L282)` comment is also incorrect.

Note how the previous paragraph talks about "support atomic access", which is a correct term here.

Reply:
True, word tearing is racing a single memory location. Good that we discovered another spec bug and could fix it in time for 25.